lib/lambda/submit/submitSplitSPA.ts

```diff
@@ -0,0 +1,117 @@
+import { getNextSplitSPAId } from "./splitSPAId";
+import { z } from "zod";
+
+const sendSubmitSplitSPAMessage = async (currentPackage: ItemResult) => {
+  const topicName = process.env.topicName as string;
+  if (!topicName) {
+    throw new Error("Topic name is not defined");
+  }
+  try {
+    const newId = await getNextSplitSPAId(currentPackage._id);
+    if (!newId) {
+      throw new Error("Error getting next Split SPA Id");
+    }
+
+    // ID and changeMade are excluded; the rest of the object has to be spread into the new package
+    const {
+      id: _id,
+      changeMade: _changeMade,
+      origin: _origin,
+      ...remainingFields
+    } = currentPackage._source;
+
+    await produceMessage(
+      topicName,
+      newId,
+      JSON.stringify({
+        id: newId,
+        idToBeUpdated: currentPackage._id,
+        ...remainingFields,
+        origin: "OneMAC",
+        changeMade: "OneMAC Admin has added a package to OneMAC.",
+        changeReason: "Per request from CMS, this package was added to OneMAC.",
+        isAdminChange: true,
+        adminChangeType: "split-spa",
+      }),
+    );
+
+    return response({
+      statusCode: 200,
+      body: { message: `New Medicaid Split SPA ${newId} has been created.` },
+    });
+  } catch (err) {
+    console.log("Error creating Split SPA: ", err);
+  }
+};
+
+const splitSPAEventBodySchema = z.object({
+  packageId: z
+    .string()
+    .regex(
+      /^[A-Z]{2}-\d{2}-\d{4}(-[A-Z0-9]{1,4})?$/,
+      "ID doesn't match format SS-YY-NNNN or SS-YY-NNNN-XXXX",
+    ),
+});
+
+export const handler = async (event: APIGatewayEvent) => {
+  if (!event.body) {
+    return response({
+      statusCode: 400,
+      body: { message: "Event body required" },
+    });
+  }
+  try {
+    const parseEventBody = (body: unknown) => {
+      return splitSPAEventBodySchema.parse(typeof body === "string" ? JSON.parse(body) : body);
+    };
+    // const { packageId } = splitSPAEventBodySchema.parse(
+    //   event.body === "string" ? JSON.parse(event.body) : event.body,
+    // );
+    let body = { packageId: "" };
+    if (typeof event.body === "string") {
+      body = JSON.parse(event.body);
+    } else {
+      body = event.body;
+    }
+    if (!body.packageId) {
+      return response({
+        statusCode: 400,
+        body: { message: "Package ID to split is required" },
+      });
+    }
+    const { packageId } = parseEventBody(event.body);
+    const currentPackage = await getPackage(packageId);
+    if (!currentPackage || currentPackage.found == false) {
+      return response({
+        statusCode: 404,
+        body: { message: "No record found for the given id" },
+      });
+    }
+
+    if (currentPackage._source.authority !== "Medicaid SPA") {
+      return response({
+        statusCode: 400,
+        body: { message: "Record must be a Medicaid SPA" },
+      });
+    }
+
+    return await sendSubmitSplitSPAMessage(currentPackage);
```

Comment:
Fun fact: you can omit `await` when returning an async function! No need to change anything if you don't feel like it